quack_app/lib/screens/homepage/home_screen.dart

```diff
@@ -17,6 +17,14 @@ class HomeScreen extends StatefulWidget {
 }
 
 class _HomeScreenState extends State<HomeScreen> {
+  @override
+  void initState() async {
+    // Re-cache updated values on view
+    super.initState();
+    await widget.menuData.getCurrentlyServing();
+    await widget.menuData.getTodaysMenu();
```

Comment:
The reason why I didn't go with what you proposed is because every time a view that needs DCT info is loaded, it will make calls to our backend to generate all DCT information, which will be heavy on our backend. 

This approach, allows for information needed to be generated once, cached, and updated as needed - i.e. (Home screen needs food data & Favorites needs favorites data. We would make 2 requests when trying to view both screens with your approach and doesn't allow for info to be cached on the user. The current approach would make the requests once and can be updated when needed).

How can we achieve this functionality:  Make requests once and update when needed
Without coupling `main.dart` to `MenuData` (considering renaming to `DctDataController` or something similar)

Reply:
I see. I think your approach works better in this case.

Reply:
@DDavid233, in the latest update with the Nav Bar I was forced to rethink the menu data process. I tried very hard to a void a `static` approach, but I think a singleton is the only way to go here. What do you think?

Reply:
@RafaelPiloto10 I think your current approach is optimal.